homeassistant/core.py

```diff
@@ -708,7 +708,17 @@ def async_run_hass_job(
                 hassjob.target = cast(Callable[..., _R], hassjob.target)
             hassjob.target(*args)
             return None
-
+        if hassjob.job_type is HassJobType.Coroutinefunction:
+            if TYPE_CHECKING:
+                hassjob.target = cast(
+                    Callable[..., Coroutine[Any, Any, _R]], hassjob.target
+                )
+            task = create_eager_task(
+                hassjob.target(*args), name=hassjob.name, loop=self.loop
+            )
+            self._tasks.add(task)
+            task.add_done_callback(self._tasks.remove)
+            return task
```

Comment:
Existing tests will hit this path and since we already have coverage for coroutetinefunctions the only difference is `create_eager_task` which also already has coverage